src/main/application/services/vendorconsent/LoadUserConsentUseCase.js

```diff
@@ -1,15 +1,28 @@
 import {inject} from '../../../core/ioc/ioc'
 import {LoadConsentService} from '../../../domain/consent/LoadConsentService'
+import {EventStatusService} from '../../../domain/service/EventStatusService'
+import {StatusRepository} from '../../../domain/status/StatusRepository'
+import {Status} from '../../../domain/status/Status'
 
 class LoadUserConsentUseCase {
-  constructor({loadConsentService = inject(LoadConsentService)} = {}) {
+  constructor({
+    loadConsentService = inject(LoadConsentService),
+    eventStatusService = inject(EventStatusService),
+    statusRepository = inject(StatusRepository)
+  } = {}) {
     this._loadConsentService = loadConsentService
+    this._eventStatusService = eventStatusService
+    this._status = statusRepository.getStatus()
   }
 
   async execute() {
-    // TODO dispatch required IAB event
     const consent = await this._loadConsentService.loadConsent()
-    return consent.toJSON()
+    const consentDto = consent.toJSON()
+    if (consentDto.valid) {
+      this._status.cmpStatus = Status.CMPSTATUS_LOADED
+      this._eventStatusService.updateTCLoaded()
```

Comment:
so every time somebody requests the User Consent, an event will be raised?

Reply:
We can discuss, and add a parameter notify, default value = false 
notify && this._eventStatusService.updateTCLoaded()
Then in TcfInitializer 
const borosTcf() = new BorosTcf()
borosTcf().loadUserConsent.execute({notify:true})
return borosTcf. 
But I am thinking to do all this in a separate PR. Moving event to initialization will break a lot of tests, and I prefer doing it in a separate PR.